src/NServiceBus.Core/BusContext.cs

```diff
@@ -6,7 +6,7 @@ namespace NServiceBus
     using NServiceBus.Pipeline;
     using NServiceBus.Unicast;
 
-    class BusContext : IBusContext
+    class BusContext : IBusContext, IBusSession
```

Comment:
I think we can't/should implement both interfaces on `BusContext`. We could run into problem when a container registers by convention bindings for all public interfaces.